src/tuner/nccl_ofi_tuner.c

```diff
@@ -0,0 +1,120 @@
+#include <stdlib.h>
+#include "nccl-headers/nvidia/tuner.h"
+#include "nccl_ofi_tuner.h"
+#include "nccl_ofi_log.h"
+
+ncclResult_t nccl_ofi_tuner_init(size_t nRanks, size_t nNodes, ncclDebugLogger_t logFunction)
+{
+	ofi_log_function = logFunction;
+
+	/*
+	 * NCCL parses these variables and applies user filters inside its
+	 * current tuner logic. Ideally, this should be done regardless of the
+	 * use of NCCL's internal tuner or an external tuner plugin. For the
+	 * time being, given the external tuner is an opt-in, detect if a user
+	 * has set one of them and bail when an external tuner is loaded.
+	 */
+	if (getenv("NCCL_ALGO") || getenv("NCCL_PROTO")) {
+		NCCL_OFI_WARN("The tuner plugin can not be loaded when explicitly choosing an algorithm or protocol with NCCL_ALGO/NCCL_PROTO");
+		return ncclInvalidUsage;
+	}
+
+	struct nccl_ofi_tuner_model_params params = {
+		.net_lat = NCCL_OFI_TUNER_NET_LATENCY,
+		.internode_bw = NCCL_OFI_TUNER_INTRANODE_BW,
+		.intranode_bw = NCCL_OFI_TUNER_INTERNODE_BW,
+		.rails = NCCL_OFI_TUNER_NET_NUM_RAILS
+	};
+
+	/*
+	 * The tuner API is missing a mechanism to pass around context after
+	 * initialization. We currently do not maintain any
+	 * communicator-specific state in the context, but if we do, multiple
+	 * invocations would overwrite data in the global context which is not
+	 * correct.
+	 */ 
+	ctx = calloc(1, sizeof(struct nccl_ofi_tuner_context));
```

Comment:
So this is the interface, but that doesn't seem like good reason not to have re-init detection.